packages/react-utils/src/helpers/dataLoaders.ts

```diff
@@ -33,6 +37,86 @@ export class OpenSRPService<T extends object = Dictionary> extends GenericOpenSR
   }
 }
 
+/** A generic FHIR service class
+ *
+ * Sample usage:
+ * -------------
+ * const serve = new FHIRServiceClass('<base url>', '<resource type>');
+ *
+ * **To list all entries of a resource (GET request)**: serve.list()
+ *
+ * **To get one resource record**: service.read('<object id>')
+ *
+ * **To create a new resource**: service.create(payload)
+ *
+ * **To update a resource record**: service.update(payload)
+ */
+export class FHIRServiceClass<T = fhirclient.FHIR.Resource> {
+  public accessTokenOrCallBack: GetAccessTokenType | string;
+  public baseURL: string;
+  public resourceType: string;
+
+  /**
+   * Constructor method
+   *
+   * @param {string} baseURL - the base FHIR URL
+   * @param {string} resourceType - FHIR resource type string
+   */
+  constructor(baseURL: string, resourceType: string) {
+    this.accessTokenOrCallBack = handleSessionOrTokenExpiry;
+    this.baseURL = baseURL;
+    this.resourceType = resourceType;
+  }
+
+  public buildQueryParams(params: URLParams | null) {
+    if (params) {
+      return `${this.resourceType}/_search?${decodeURIComponent(queryString.stringify(params))}`;
+    }
+    return this.resourceType;
+  }
+
+  private buildState(accessToken: string) {
+    return {
+      serverUrl: this.baseURL,
+      tokenResponse: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        access_token: accessToken,
+      },
+    };
+  }
+
+  public async create(payload: T) {
+    const accessToken = await OpenSRPService.processAcessToken(this.accessTokenOrCallBack);
+    const serve = FHIR.client(this.buildState(accessToken));
+    return serve.create(payload);
+  }
+
+  public async update(payload: T) {
+    const accessToken = await OpenSRPService.processAcessToken(this.accessTokenOrCallBack);
+    const serve = FHIR.client(this.buildState(accessToken));
+    return serve.update(payload);
+  }
+
+  public async list(params: URLParams | null = null) {
+    const accessToken = await OpenSRPService.processAcessToken(this.accessTokenOrCallBack);
+    const queryStr = this.buildQueryParams(params);
+    const serve = FHIR.client(this.buildState(accessToken));
+    return serve.request(queryStr);
+  }
+
+  public async read(id: string) {
```

Comment:
should we unify read and list to one request method?

Reply:
Trying to maintain a similar work flow as the OpenSRP service class so that a new developer can implement both services easily